models/wrf_hydro/noah_hydro_mod.f90

```diff
@@ -654,10 +654,8 @@ subroutine getChannelGridCoords(filename, iunit, numdims, ncstart, nccount)
 end do
 deallocate(CH_NETRT_in, LAKE_MSKRT_in, DIRECTION_in, ELRT_in)
 
-! This replaces a double for loop that counts NLINKS which can be removed from the
-! code inserted below.
 ! subset to the 1D channel network as presented in the hydro restart file.
-n_link = sum(CH_NETRT*0+1, mask = CH_NETRT >= 0)
+n_link = count(CH_NETRT>=0)
```

Comment:
see #599  need to confirm that the intent of the original code is to count all the elements in CH_NETRT that are >0

Reply:
I made a comment on the issue addressed to Moha; I will send him an email about it

Reply:
Confirmed that the original intent behind this line is to count all non-negative elements in the array